Microsoft.Azure.Cosmos/src/Linq/TranslationContext.cs

```diff
@@ -81,10 +81,25 @@ public TranslationContext(CosmosLinqSerializerOptions linqSerializerOptions, IDi
             this.collectionStack = new List<Collection>();
             this.CurrentQuery = new QueryUnderConstruction(this.GetGenFreshParameterFunc());
             this.subqueryBindingStack = new Stack<SubqueryBinding>();
-            this.LinqSerializerOptions = linqSerializerOptions;
             this.Parameters = parameters;
-            this.MemberNames = new MemberNames(linqSerializerOptions);
-            this.CosmosLinqSerializer = new DefaultCosmosLinqSerializer();
+
+            if (linqSerializerOptionsInternal?.CustomCosmosLinqSerializer != null)
+            {
+                this.CosmosLinqSerializer = new CustomCosmosLinqSerializer(linqSerializerOptionsInternal.CustomCosmosLinqSerializer);
+                this.MemberNames = new MemberNames(new CosmosLinqSerializerOptions());
+            }
+            else if (linqSerializerOptionsInternal?.CosmosLinqSerializerOptions != null)
+            {
+                CosmosLinqSerializerOptions linqSerializerOptions = linqSerializerOptionsInternal?.CosmosLinqSerializerOptions ?? new CosmosLinqSerializerOptions();
+
+                this.CosmosLinqSerializer = new DefaultCosmosLinqSerializer(linqSerializerOptions.PropertyNamingPolicy);
+                this.MemberNames = new MemberNames(linqSerializerOptions);
+            }
+            else
+            {
+                this.CosmosLinqSerializer = TranslationContext.DefaultLinqSerializer;
+                this.MemberNames = DefaultMemberNames;
```

Comment:
nit: Classname prefix

```suggestion
                this.MemberNames = TranslationContext.DefaultMemberNames;
```